### ISSUE TITLE
Option to deploy an existing release in OctopusDeploy 

### ISSUE THREAD

**Issue by trailmax** [cake-build/cake#1228]
Once you crated a release in Octopus Deploy through Cake, there is no option to deploy that release. I've got a draft code that can do that, but I see there are a few issues around OctopusDeploy:  #1022, pending PR #1218.

I'm not sure what is the conclusion of discussion about duplicating parameters for server and apikey in alias calls and settings object. So here is my proposal:

        [CakeMethodAlias]
        public static void OctoDeployRelease(this ICakeContext context, string server, string apiKey, string projectName, string deployTo, string releaseNumber, DeployReleaseSettings settings)
        {
            // do deploy
        }

        [CakeMethodAlias]
        public static void OctoDeployRelease(this ICakeContext context, string server, string apiKey, string projectName, string deployTo,  DeployReleaseSettings settings)
        {
            OctoDeployRelease(context, server, apiKey, projectName, deployTo, "Latest", settings);
        }

My only concern is #1022 - Should I base the settings object on common `OctopusDeploySettings` while it still has Server and ApiKey parameters?

**Comment by patriksvensson** [comment]
@trailmax Looks like release number isn't mandatory, so it should be on the settings class. `DeployReleaseSettings` is also too generic, so would be better if you called that `OctoDeployReleaseSettings` to keep consistency with other tools.

```
[CakeMethodAlias]
    public static void OctoDeployRelease(this ICakeContext context, string server, string apiKey, string projectName, string deployTo,  DeployReleaseSettings settings)
    {
        OctoDeployRelease(context, server, apiKey, projectName, deployTo, "Latest", settings);
    }
```

**Comment by patriksvensson** [comment]
@trailmax `projectName` and `deployTo` is required by the underlying tool as well to function?

**Comment by trailmax** [comment]
Yes, it would not now what to do without these parameters - see http://docs.octopusdeploy.com/display/OD/Deploying+releases

**Comment by trailmax** [comment]
@patriksvensson ReleaseNumber is not optional - it can be "Latest" or a version, but it must be there. One of the ways to do it - have default settings object set up with "Latest", but then option to override it. I almost like it, but then what version are you deploying is quite important and I can see that every call to deploy will end up with settings object specifying the exact release. So I'm in favour of having it as a mandatory param outside of the settings object.

**Comment by patriksvensson** [comment]
@trailmax Ok, but then I think it should be up to the user to pass `latest` and only have one alias since this convention potentially could be changed, and we would keep as close to the underlying tool as possible.

**Comment by trailmax** [comment]
@patriksvensson Sounds good. I won't do the override then.

**Comment by patriksvensson** [comment]
@trailmax Perfect! :+1:

**Comment by devlead** [comment]
Fixed by #1229
